planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp

```diff
@@ -1400,15 +1249,307 @@ std::vector<lanelet::ConstPoint3d> getBoundWithIntersectionAreas(
+
+    // extract line strings between start_idx and end_idx.
+    const size_t start_idx = std::distance(polygon.begin(), start_point_itr);
+    const size_t end_idx = std::distance(polygon.begin(), end_point_itr);
+
+    return extractBoundFromPolygon(polygon, start_idx, end_idx, is_clockwise_iteration);
+  };
+
+  const auto get_bound_edge = [&ego_pose, &is_driving_freespace, &is_left](
+                                const auto & bound, const auto is_inside) {
+    if (!is_driving_freespace) {
+      return bound;
+    }
+
+    const auto p_offset = tier4_autoware_utils::calcOffsetPose(
+      ego_pose, (is_inside ? -100.0 : 100.0), (is_left ? 0.1 : -0.1), 0.0);
+
+    std::vector<lanelet::ConstPoint3d> ret;
+    for (size_t i = 1; i < bound.size(); ++i) {
+      const auto intersect = tier4_autoware_utils::intersect(
+        ego_pose.position, p_offset.position, toGeomMsgPt(bound.at(i - 1)),
+        toGeomMsgPt(bound.at(i)));
+
+      ret.push_back(bound.at(i - 1));
+
+      if (intersect.has_value()) {
+        ret.emplace_back(
+          lanelet::InvalId, intersect.value().x, intersect.value().y, intersect.value().z);
+        break;
+      }
+    }
+
+    return ret;
+  };
+
+  std::vector<lanelet::ConstPoint3d> expanded_bound;
+
+  if (original_bound_itr->id() != original_bound_rev_itr->id()) {
+    const auto polygon_bound =
+      extract_bound_from_polygon(original_bound_itr->id(), original_bound_rev_itr->id());
+
+    expanded_bound.insert(expanded_bound.end(), original_bound.begin(), original_bound_itr);
+    expanded_bound.insert(expanded_bound.end(), polygon_bound.begin(), polygon_bound.end());
+    expanded_bound.insert(
+      expanded_bound.end(), std::next(original_bound_rev_itr).base(), original_bound.end());
+
+  } else if (boost::geometry::within(
+               to2D(original_bound.front().basicPoint()), to2D(polygon).basicPolygon())) {
+    auto polygon_bound =
+      extract_bound_from_polygon(other_side_bound_itr->id(), original_bound_itr->id());
+    std::reverse(polygon_bound.begin(), polygon_bound.end());
+    auto bound_edge = get_bound_edge(polygon_bound, true);
+    std::reverse(bound_edge.begin(), bound_edge.end());
+
+    expanded_bound.insert(expanded_bound.end(), bound_edge.begin(), bound_edge.end());
+    expanded_bound.insert(expanded_bound.end(), original_bound_itr, original_bound.end());
+
+  } else if (boost::geometry::within(
+               to2D(original_bound.back().basicPoint()), to2D(polygon).basicPolygon())) {
+    const auto polygon_bound =
+      extract_bound_from_polygon(original_bound_itr->id(), other_side_bound_itr->id());
+    const auto bound_edge = get_bound_edge(polygon_bound, false);
+
+    expanded_bound.insert(expanded_bound.end(), original_bound.begin(), original_bound_itr);
+    expanded_bound.insert(expanded_bound.end(), bound_edge.begin(), bound_edge.end());
+
+  } else {
+    expanded_bound = original_bound;
+  }
```

Comment:
better to add explanation to each patter

Reply:
I believe it greatly helps code readability and reduce complexity if you use enums to describe what is happening and then use a switch for each case: 

something like
 
`enum Classification { CASE_A, CASE_B, CASE_C, CASE_D };

const auto classifyBound = [&]() -> Classification
{
  if (original_bound_itr->id() != original_bound_rev_itr->id()) return Classification::CASE_A;
  if (boost::geometry::within(to2D(original_bound.front().basicPoint())))
    return Classification::CASE_B;
  if (boost::geometry::within(to2D(original_bound.back().basicPoint())))
    return Classification::CASE_C;
  return Classification::CASE_D;
};

cacse = classifyBound()

switch(case){

...
}
`

If you change the names of the cases for a descriptive name and then use a switch I believe it would be more descriptive without having to write any comments

Reply:
I fixed implementation for readability in https://github.com/autowarefoundation/autoware.universe/pull/6001/commits/28badef8ecedf6510ba7e155ae508666a80ae9b0.